cmd/admin-bucket-quota.go

```diff
@@ -0,0 +1,159 @@
+	QuotaType string `json:"type,omitempty"`
+}
+
+func (q quotaMessage) String() string {
+	if q.op == "set" {
+		return console.Colorize("QuotaMessage",
+			fmt.Sprintf("Successfully set bucket quota of %s with %s type on `%s`", humanize.IBytes(q.Quota), q.QuotaType, q.Bucket))
+	}
+	return console.Colorize("QuotaInfo",
+		fmt.Sprintf("Bucket `%s` has %s quota of %s", q.Bucket, q.QuotaType, humanize.IBytes(q.Quota)))
+}
+
+func (q quotaMessage) JSON() string {
+	q.Status = "success"
+	jsonMessageBytes, e := json.MarshalIndent(q, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(jsonMessageBytes)
+}
+
+var adminBucketQuotaCmd = cli.Command{
+	Name:   "quota",
+	Usage:  "Set/Get bucket quota",
+	Action: mainAdminBucketQuota,
+	Before: setGlobalsFromContext,
+	Flags:  append(adminQuotaFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+   {{.HelpName}} - {{.Usage}}
+ 
+USAGE:
+   {{.HelpName}} TARGET USERNAME
+ 
+UNITS
+   quota accepts human-readable case-insensitive number
+   suffixes such as "k", "m", "g" and "t" referring to the metric units KB,
+   MB, GB and TB respectively. Adding an "i" to these prefixes, uses the IEC
+   units, so that "gi" refers to "gibibyte" or "GiB". A "b" at the end is
+   also accepted. Without suffixes the unit is bytes.
+ 
+FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+EXAMPLES:
+   1. Display bucket quota configured for "mybucket".
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket
+	
+   2. Set FIFO quota for a bucket "mybucket".
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket 64kB --fifo
+`,
+}
+
+// checkAdminBucketQuotaSyntax - validate all the passed arguments
+func checkAdminBucketQuotaSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 2 {
+		cli.ShowCommandHelpAndExit(ctx, "quota", 1) // last argument is exit code
+	}
+
+	if ctx.IsSet("hard") && ctx.IsSet("fifo") {
+		fatalIf(errInvalidArgument(), "Only one of --hard or --fifo flags can be set")
+	}
+	if (ctx.IsSet("hard") || ctx.IsSet("fifo")) && len(ctx.Args()) != 2 {
+		fatalIf(errInvalidArgument().Trace(ctx.Args()...), "please specify bucket and quota")
+	}
+}
+
+// mainAdminBucketQuota is the handler for "mc admin bucket quota" command.
+func mainAdminBucketQuota(ctx *cli.Context) error {
+	checkAdminBucketQuotaSyntax(ctx)
+
+	console.SetColor("QuotaMessage", color.New(color.FgGreen))
+	console.SetColor("QuotaInfo", color.New(color.FgBlue))
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Unable to initialize admin connection.")
+	fifoFlag := ctx.Bool("fifo")
+	hardFlag := ctx.Bool("hard")
+	_, targetURL := url2Alias(ctx.Args()[0])
+	if fifoFlag || hardFlag && len(ctx.Args()) == 2 {
+		qType := madmin.FIFOQuota
+		if hardFlag {
+			qType = madmin.HardQuota
+		}
+		quota, err := humanize.ParseBytes(ctx.Args()[1])
```

Comment:
Suggestion (optional): I propose that we move parsing of quota size into `checkAdminBucketQuotaSyntax(ctx)`.  The idea is to keep all the validation code in one place.